rfcs/search_relevancy_sandbox/20230331-project_proposal_search_relevancy_sandbox.md

```diff
@@ -0,0 +1,145 @@
+4. A mechanism by which maintainers can easily deploy & iterate on new
+   Elasticsearch indexing configurations
+5. A mechanism by which maintainers can easily point the staging Elasticsearch
+   index aliases (currently `image` and `audio`) to one of the above.
+
+Many of these mechanisms will likely be built in Airflow.
+
+### Proportional by provider
+
+The proportional-by-provider index will be a subset of the production data,
+constructed so that the number of records by provider is roughly proportional to
+the percentages per provider that exist in production. This index will allow us
+to create tests down the line which might assess result relevancy in an
+automated way, i.e. based on how many results from a particular provider appear
+in results (see the
+[provider tally script](https://github.com/WordPress/openverse/pull/397) for
+examples of these results). This could be done by:
+
+1. Querying the `/stats` endpoint of the production API (e.g.
+   https://api.openverse.engineering/v1/images/stats/).
+2. Calculating percentage-per-provider.
+3. Computing the number of results per provider based on these percentages based
+   on the condition that the smallest provider has a minimum `N` results (e.g. a
+   minimum of 10 results).
+4. Pulling a number of results based on the previous step per-provider either
+   into a Postgres table or an Elasticsearch index (the specifics of this step
+   will be left for the implementation plan).
+
+This index will allow us to assess how changes to Elasticsearch index or the API
+will affect results using a smaller total result count. It will also allow us to
+test API results from all currently available providers.
+
+### Production data volume
+
+The production-data-volume index will be an index of similar size to production.
+This is how the data refresh is currently set up and does not require any
+additional work to set up.
+
+This index will allow us to test how queries and indexing operations perform at
+a data volume consistent with production.
+
+## Success
+
+<!-- How do we measure the success of the project? How do we know our ideas worked? -->
+
+This project can be considered a success once any maintainer can make
+adjustments to the staging API database & Elasticsearch index based on the
+requirements described above.
+
+## Participants and stakeholders
+
+<!-- Who is working on the project and who are the external stakeholders, if any? Consider the lead, implementers, designers, and other stakeholders who have a say in how the project goes. -->
+
+- Lead: @AetherUnbound
+- Implementation:
+  - @AetherUnbound
+  - TBD
+- Stakeholders:
+  - Openverse Team
+
+## Infrastructure
+
+<!-- What infrastructural considerations need to be made for this project? If there are none, say so explicitly rather than deleting the section. -->
+
+This project may require infrastructure work necessary to assist with the rapid
+iteration of the data refresh server. Further details will be determined in the
```

Comment:
Does "data refresh server" mean the ingestion server project?

Reply:
Yes it does, I was torn trying to determine which to put down since the _process_ was renamed to "data refresh", but all our code and server references are to "ingestion server" so I'll change it back.

Reply:
Yeah, unfortunately with already unclear names I think it's worth sticking with the specific one until they're actually changed in code, otherwise it's quite confusing :disappointed:. It's worth noting that "ingestion server" _does_ make sense from an API perspective: the service ingests catalogue data into ES and the API database. Maybe "API ingestion server" would be a good interim name?

Reply:
I think "ingestion server" is fine to remain as-is here. The catalog _does_ ingestion but I wouldn't consider it an "ingestion server", and we refer to the "data refresh" process separately anyway.